src/test/java/org/broadinstitute/hellbender/engine/GATKToolUnitTest.java

```diff
@@ -281,4 +321,207 @@ public void testBestSequenceDictionary_fromNothing() throws Exception {
+            final boolean createIndex,
+            final boolean createMD5) throws IOException {
+
+        // verify lenient==false is honored by writing a bad attribute
+        final TestGATKToolWithVariants tool = new TestGATKToolWithVariants();
+        final File outputFile = setupVCFWriter(inputFile, outputExtension, tool, createIndex, createMD5, false);
+
+        try (VariantContextWriter writer = tool.createVCFWriter(outputFile)) {
+            writeHeaderAndBadVariant(writer); // throws due to bad attribute
+        }
+    }
+
+    @CommandLineProgramProperties(
+            summary = "TestGATKVariantToolWithNoSequenceDictionary",
+            oneLineSummary = "TestGATKVariantToolWithNoSequenceDictionary",
+            programGroup = TestProgramGroup.class
+    )
+    private static final class TestGATKVariantToolWithNoSequenceDictionary extends GATKTool{
+        @Argument(fullName="out", shortName="out", doc="Input variants", optional=true)
+        public File out;
+
+        @Override
+        public SAMSequenceDictionary getBestAvailableSequenceDictionary() {return null;}
+
+        @Override
+        public boolean requiresFeatures() {return true;}
+
+        @Override
+        public void traverse() {}
+    }
+
+    @Test(dataProvider = "createVCFWriterData")
+    public void testCreateVCFWriterWithNoSequenceDictionary(
+            final File inputFile,
+            final String outputExtension,
+            final String indexExtension,
+            final boolean createIndex,
+            final boolean createMD5) throws IOException
+    {
+        // verify that a null sequence dictionary still results in a file, but with no index
+        final TestGATKVariantToolWithNoSequenceDictionary tool = new TestGATKVariantToolWithNoSequenceDictionary();
+        final File outputFile = setupVCFWriter(inputFile, outputExtension, tool, createIndex, createMD5, false);
+
+        final VariantContextWriter writer = tool.createVCFWriter(outputFile);
+        writer.close();
+
+        final File outFileIndex = new File(outputFile.getAbsolutePath() + indexExtension);
+        final File outFileMD5 = new File(outputFile.getAbsolutePath() + ".md5");
+
+        Assert.assertTrue(outputFile.exists());
+        Assert.assertEquals(outFileIndex.exists(), false); // always false with no seq dictionary
+        Assert.assertEquals(outFileMD5.exists(), createMD5);
+    }
+
+    private File setupVCFWriter(
+            final File inputFile,
+            final String outputExtension,
+            final GATKTool tool,
+            final boolean createIndex,
+            final boolean createMD5,
+            final boolean lenient) throws IOException
+    {
+        final File tmpDir = createTempDir("createVCFTest");
+        final File outputFile = new File(tmpDir.getAbsolutePath(), "createVCFTest" + outputExtension);
+
+        List<String> args = new ArrayList<>();
+        args.add("--input"); args.add(inputFile.getCanonicalPath());
```

Comment:
we have some ArgumentsBuilder convenience methods to do this less verbosely if you prefer